### PR TITLE
Reconcile feature status with https://github.com/istio/community/edit/master/FEATURE-LIFECYCLE.md

### PR DESCRIPTION
**Pull request by jacob-delgado**
Copy of contents from https://github.com/istio/community/edit/master/FEATURE-LIFECYCLE.md

Follow-up to https://github.com/istio/community/pull/495

With Istio 1.9 having the focus of being a stability release it is proper to define what that means to end users and developers on the project.

This was mentioned at the TOC a few months back, but with the definition
of a feature's lifecycle/status being in 2 places results in user and
developer confusion as far as the expectations for a feature. 

Before we merge we need community and TOC guidance if they wish to continue with this approach.

We may not wish to keep each row, but the current feature status page leaves something to be desired.

[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by nrjpoddar** [comment]
I think we reviewed contents of https://github.com/istio/community/edit/master/FEATURE-LIFECYCLE.md extensively so the main question here is if this content is suitable for end users, correct?

**Comment by nrjpoddar** [comment]
I like this change as the table is quite comprehensive (may be a bit verbose) but adds clarity if I am an end user as to what's expected from any Istio feature.

**Comment by jacob-delgado** [comment]
> I think we reviewed contents of [istio/community/edit/master/FEATURE-LIFECYCLE.md](https://github.com/istio/community/edit/master/FEATURE-LIFECYCLE.md) extensively so the main question here is if this content is suitable for end users, correct?

Yes. There is no mention of what experimental is, so even though we don't mention the experimental features, istio.io contains references to it. Having 2 places of truth confuses devs and end users. End users will, at times, need to use alpha/beta feature, but it should be clear that their expectations match ours. When it's not is when people get frustrated.

**Review by nrjpoddar — APPROVED** [review]

**Comment by ericvn** [comment]
@jacob-delgado Just waiting on spelling fixes.

**Comment by ericvn** [comment]
Spelling issues:
```
    ./en/about/feature-stages/index.md
       24 |                             | Perf (latency/scale) is quantified 
       27 | rsion name contains dev (e.g. v1dev1)                              
       27 | sion name contains beta (e.g. v2beta3)                              
       27 |          | The API version is vX where X is an integer (e.g. v 
       28 | e available in the main Istio repo. The feature may or may not a 
       28 | ure is committed to the Istio repo The feature appears in an off 
       32 | | Upgradeability        | The schema and seman 
       32 | ration objects in an existing istio installation API versions may 
       33 |                             | Superset of Beta, including tests for  
       34 |     | Because the feature has e2e tests, enabling the feature s 
       35 | ve versions (e.g. v1beta1 and v1beta2; or v1beta2 and v1) for at le 
       35 | (e.g. v1beta1 and v1beta2; or v1beta2 and v1) for at least one supp 
       36 | onments, due to complexity of upgradeability and lack of long-term support 

>> 13 spelling errors found in 623 files
```

File: content/en/about/feature-stages/index.md [hunk 1]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...ons for backwards compatibility. Developers are encouraged to increment the API version when schema or semantics change in an incompatible way | The schema and semantics of an alpha feature may change in a later software release, without any provision for preserving configuration objects in an existing Istio installation API versions may increment faster than the monthly release cadence and the developer need not maintain multiple versions Developers should increment the API version when schema or semantics change in an incompatible way                                      | The schema and semantics may change in a later software release When this happens, an upgrade path will be documented In some cases, objects will be automatically converted to the new version In other cases, a manual upgrade may be necessary A manual upgrade may require downtime for anything relying on the new feature, and may require manual conversion of objects to the new version When manual conversion is necessary, the project will provide documentation on the process | Only strictly compatible changes are allowed in subsequent software releases                                                               |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:32]: I wouldn't use quotes around Upgradability as it becomes the only highlighted entry. I suspect it's because of a spelling issue. I would add it to .spelling if that is the case

File: content/en/about/feature-stages/index.md [hunk 2]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...                                                                                                                                               | No guarantees on backward compatibility. The feature may be removed at any time without notice.                                                                                                                                                                                                                                                                                                                                                    | The overall feature will not be dropped, though details may change.                                                                                                                                                                                                                                                                                                                                                                                                                         | The feature will appear in released software for many subsequent versions.                                                                 |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:25]: Here you mention the feature won't be dropped (only column with no drop), but we talk about deprecation on the next row.

File: content/en/about/feature-stages/index.md [hunk 3]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
... turn on dev features.                                                                                                                         | The feature is committed to the Istio repository The feature appears in an official Istio release The feature requires explicit user action to use, e.g. a flag flip, use of a config resource, an explicit installation action, or an API being called. When a feature is disabled it must not affect system stability.                                                                                                                           | In official Istio releases Enabled by default                                                                                                                                                                                                                                                                                                                                                                                                                                               | Same as Beta                                                                                                                               |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:28]:
```suggestion
| Availability          | The feature may or may not be available in the main Istio repository. The feature may or may not appear in an Istio release. If it does appear in an Istio release it will be disabled by default. The feature requires an explicit flag to enable in addition to any configuration required to use the feature, in order to turn on dev features.                                                                                                                         | The feature is committed to the Istio repository. The feature appears in an official Istio release. The feature requires explicit user action to use, e.g. a flag flip, use of a config resource, an explicit installation action, or an API being called. When a feature is disabled it must not affect system stability.                                                                                                                           | In official Istio releases Enabled by default                                                                                                                                                                                                                                                                                                                                                                                                                                               | Same as Beta                                                                                                                               |
```

File: content/en/about/feature-stages/index.md [hunk 4]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...                                                                                                                                               | None, feature can be removed at any time                                                                                                                                                                                                                                                                                                                                                                                                           | Weak, feature can be removed with 3 months of advanced notice                                                                                                                                                                                                                                                                                                                                                                                                                               | Strong, feature can be removed with 1 year of advanced notice, but there will usually be a supported upgrade path to a replacement feature |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:26]:
```suggestion
| Deprecation Policy    | None, feature can be removed at any time.                                                                                                                                                                                                                                                                                                                                                                                                                                   | None, feature can be removed at any time.                                                                                                                                                                                                                                                                                                                                                                                                           | Weak, feature can be removed with 3 months of advanced notice.                                                                                                                                                                                                                                                                                                                                                                                                                               | Strong, feature can be removed with 1 year of advanced notice, but there will usually be a supported upgrade path to a replacement feature. |
```

Comment (ericvn) [reply content/en/about/feature-stages/index.md:26]: There is a lot of inconsistenty with periods.  https://deploy-preview-8903--preliminary-istio.netlify.app/latest/about/feature-stages/ for current deployment. Sometimes it missing in some columns of a row. Sometime for a whole column.

File: content/en/about/feature-stages/index.md (hunk 1 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:32]: Also we mention a monthly cadence?

File: content/en/about/feature-stages/index.md [hunk 5]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...                                                                                                                                               | The feature is covered by unit tests and integration tests where the feature is enabled and the tests are non-flaky. Tests may not cover all corner cases, but the most common cases have been covered. Testing code coverage >= 70%. When the feature is disabled it does not regress performance of the system.                                                                                                                                  | Integration tests cover edge cases as well as common use cases. Integration tests cover all issues reported on the feature. The feature has end-to-end tests covering the samples/tutorials for the feature. Test code coverage is >= 80%.                                                                                                                                                                                                                                                  | `Superset` of Beta, including tests for any issues discovered during Beta. Test code coverage is >= 90%.                                   |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:33]: Similar issue with superset to upgradability.

Comment (ericvn) [reply content/en/about/feature-stages/index.md:33]: I don't know where we actually measure code coverage.

File: content/en/about/feature-stages/index.md [hunk 6]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...                                                                                                                                               | Because the feature is relatively new, and may lack complete end-to-end tests, enabling the feature via a flag might expose bugs which destabilize Istio (e.g. a bug in a control loop might rapidly create excessive numbers of objects, exhausting API storage).                                                                                                                                                                                 | Because the feature has `e2e` tests, enabling the feature should not create new bugs in unrelated features. Because the feature is relatively new, it may have minor bugs.                                                                                                                                                                                                                                                                                                                  | High. The feature is well tested and stable and reliable for all uses.                                                                     |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:34]: Same comment for e2e. Why do we say `High` in the stable column but nothing in the others?

File: content/en/about/feature-stages/index.md [hunk 7]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
...
...                                                                                                                                               | There is no commitment from Istio to complete the feature The feature may be dropped entirely in a later software release                                                                                                                                                                                                                                                                                                                          | Istio commits to complete the feature, in some form, in a subsequent Stable version Typically this will happen within 3 months, but sometimes longer Releases should simultaneously support two consecutive versions (e.g. v1beta1 and `v1beta2`; or `v1beta2` and v1) for at least one supported release cycle (typically 3 months) so that users have enough time to upgrade and migrate resources.                                                                                       | The feature will continue to be present for many subsequent releases.                                                                      |
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:35]: v1betax, same issue as above.

Comment (jacob-delgado) [reply content/en/about/feature-stages/index.md]: Fixed

File: content/en/about/feature-stages/index.md (hunk 1 above)
Comment (jacob-delgado) [reply content/en/about/feature-stages/index.md]: Removed the notion of a monthly cadence. This was taken from https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE.md

It obviously needs more work before we can merge this in.

File: .spelling [hunk 8]
```diff
@@ -440,7 +441,10 @@ linters
 list_by_publishdate
 liveness
 localhost
+<<<<<<< HEAD
```
Comment (ericvn) [inline .spelling:444]: Remove items related to rebase.

File: .spelling [hunk 9]
```diff
@@ -722,6 +727,7 @@ Unmarshaling
 unnormalized
 unsampled
 untrusted
+Upgradeability
```
Comment (ericvn) [inline .spelling:730]: Needs to be lower case as well.

File: content/en/about/feature-stages/index.md [hunk 10]
```diff
@@ -16,17 +16,29 @@ within the project, not to the project as a whole. Here is a high level descript
 
 ## Feature phase definitions
 
-|            | Alpha      | Beta         | Stable
```
Comment (ericvn) [inline content/en/about/feature-stages/index.md:19]: Line 14 contains a list of phases, bus is missing Experimental in the list. Might be better to just remove the list of phases.

File: content/en/about/feature-stages/index.md (hunk 2 above)
Comment (jacob-delgado) [reply content/en/about/feature-stages/index.md]: I think that's an issue we could file in istio/community and the TSC can update that wording.

**Review by ericvn — APPROVED** [review]
